src/sentry/integrations/slack/utils.py

```diff
@@ -269,3 +269,45 @@ def parse_link(url):
     parsed_path += "/" + str(url_parts[4])
 
     return parsed_path
+
+
+def get_emails_by_user(organization):
```

Comment:
This helper can still be simplified into one DB query.
For example, you can add a manager to src/sentry/models/useremail.py:
```
class UserEmailManager(BaseManager):
    def get_for_organization(self, organization: Any) -> QuerySet:
        return self.filter(
            user__sentry_orgmember_set__organization=organization
        )

    def get_emails_by_user(self, organization: Any) -> Mapping[Any: Set[str]]:
        emails_by_user = defaultdict(set)
        user_emails = self.get_for_organization(organization).select_related("user") 
        for entry in user_emails:
            emails_by_user[entry.user].add(entry.email) 
        return emails_by_user 
```
and use it as `UserEmail.objects.get_emails_by_user(organization)`